File: v-lang/lang.java/v-utils/src/main/java/org/example/json/GsonUtils.java
package org.example.json;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import lombok.experimental.UtilityClass;

@UtilityClass
public class GsonUtils {

    private final Gson gson = new Gson();

    public <T> T json2Object(String json, Class<T> clazz) {
        return gson.fromJson(json, clazz);
    }

    public <T> T json2Object(String json, TypeToken<T> typeToken) {
        return gson.fromJson(json, typeToken);
    }

    public String object2Json(Object obj) {
        return gson.toJson(obj);
    }
}

File: v-lang/lang.kt/v-utils/u-json/src/main/kotlin/utils/GsonUtils.kt
package org.example.utils

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

object GsonUtils {

    private val gson = Gson()

    fun <T> json2Object(json: String, clazz: Class<T>): T = gson.fromJson(json, clazz)

    fun <T> json2Object(json: String, typeToken: TypeToken<T>): T = gson.fromJson(json, typeToken)

    fun object2Json(obj: Any): String = gson.toJson(obj)
}